src/hdmf_zarr/backend.py

```diff
@@ -718,7 +754,7 @@ def resolve_ref(self, zarr_ref):
             source_file = str(zarr_ref['source'])
         # Resolve the path relative to the current file
         if not self.is_remote():
-            source_file = os.path.abspath(os.path.join(self.source, source_file))
+            source_file = os.path.abspath(source_file)
```

Comment:
Can you clarify why `self.source` does not need to appear here? Isn't `source_file` a path relative to this file (i.e., `self.source`)?